libs/blocks/interlinks/interlinks.js

```diff
@@ -0,0 +1,104 @@
+import {
+  getRootPath,
+} from '../../scripts/scripts.js';
+
+/**
+ * Checks if a given match intersects with an existing match
+ * before adding it to the list of matches. In case of an
+ * intersection, the more specific (i.e. longer) match wins.
+ * @param {array} matches The existing matches
+ * @param {object} contender The match to check and add
+ * @param {number} maxMatches The maximum number of matches
+ */
+export function checkAndAddMatch(matches, contender, maxMatches) {
+  const collisions = matches
+    // check for intersections
+    .filter((match) => {
+      return !(contender.end < match.start || contender.start > match.end);
+    });
+  if (collisions.length === 0 && matches.length < maxMatches) {
+    // no intersecting existing matches, add contender if max not yet reached
+    matches.push(contender);
+  }
+}
+
+/**
+ * Loops through a list of keywords and looks for matches in the article text.
+ * The first occurrence of each keyword will be replaced with a link and tracking added.
+ */
+export default async function interlink() {
+  const articleBody = document.querySelector('main');
+  const resp = await fetch(`${getRootPath()}/keywords.json`);
+  if (articleBody && resp.ok) {
```

Comment:
minor: use early return instead of large if block:
```suggestion
  if (!(articleBody && resp.ok)) return;
```